internal/manifests/manifests.go

```diff
@@ -540,12 +540,12 @@ func (m *Manifests) validateUserSuppliedManifest(ctx context.Context, clusterID
 		if !json.Valid(manifestContent) {
 			return m.prepareAndLogError(ctx, http.StatusBadRequest, errors.Errorf("Manifest content of file %s for cluster ID %s has an illegal JSON format", fileName, string(clusterID)))
 		}
-	} else if strings.HasPrefix(extension, ".patch") && (strings.Contains(fileName, ".yaml.patch") || strings.Contains(fileName, ".yml.patch")) {
+	} else if strings.Contains(fileName, ".yaml.patch") || strings.Contains(fileName, ".yml.patch") {
 		if err := isValidYaml(manifestContent); err != nil {
```

Comment:
shouldn't we validate that the file is a valid yaml patch instead? https://github.com/openshift/assisted-service/blob/master/internal/common/common.go#L482

Reply:
Agreed!
I have changed this.